src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/source_filters_table/components/table/table.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiInMemoryTable,
+  EuiFieldText,
+  EuiButtonIcon,
+  keyCodes,
+  RIGHT_ALIGNMENT,
+} from '@elastic/eui';
+
+export class Table extends Component {
+  static propTypes = {
+    indexPattern: PropTypes.object.isRequired,
+    items: PropTypes.array.isRequired,
+    deleteFilter: PropTypes.func.isRequired,
+    fieldWildcardMatcher: PropTypes.func.isRequired,
+    saveFilter: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      editingFilterId: null,
+      editingFilterValue: null,
+    };
+  }
+
+  startEditingFilter = (id, value) => this.setState({ editingFilterId: id, editingFilterValue: value })
+  stopEditingFilter = () => this.setState({ editingFilterId: null })
+  onEditingFilterChange = e => this.setState({ editingFilterValue: e.target.value })
+
+  onEditFieldKeyPressed = ({ charCode }) => {
```

Comment:
I was having trouble getting the ESCAPE key to cancel editing. I changed `charCode` to `keyCode` and changed the `onKeyPress` handler to `onKeyDown` (see below), and it worked:

```js
  onEditFieldKeyPressed = ({ keyCode }) => {
    if (keyCodes.ENTER === keyCode) {
      this.props.saveFilter({
        filterId: this.state.editingFilterId,
        newFilterValue: this.state.editingFilterValue,
      });
      this.stopEditingFilter();
    }
    if (keyCodes.ESCAPE === keyCode) {
      this.stopEditingFilter();
    }
  }
```

Reply:
Yes, sounds good!